src/Main.elm

```diff
@@ -25,40 +143,939 @@ init =
+stringToBool maybeStr =
+    case maybeStr of
+        Just _ ->
+            Decode.succeed True
+
+        Nothing ->
+            Decode.succeed False
+
+
+decodePrefMethod : Decoder PreferredContactMethod
+decodePrefMethod =
+    Decode.string |> Decode.andThen stringToPrefMethod
+
+
+stringToPrefMethod : String -> Decoder PreferredContactMethod
+stringToPrefMethod str =
+    case str of
+        "email" ->
+            Decode.succeed Email
+
+        "address" ->
+            Decode.succeed Post
 
+        "phone" ->
+            Decode.succeed Phone
 
-header : String -> Html msg
-header text =
-    Html.span [ Attrs.class "p-2 text-5xl font-extrabold text-transparent bg-clip-text bg-gradient-to-br from-slate-400 to-slate-800" ]
-        [ Html.text text ]
+        _ ->
+            Decode.fail <| "Invalid preferred method: " ++ str
 
 
-subheader : String -> Html msg
-subheader text =
-    Html.span [ Attrs.class "p-2 text-2xl font-extrabold text-slate-800" ]
-        [ Html.text text ]
+decodeTagTuple : Decoder ( TagName, String )
+decodeTagTuple =
+    Decode.succeed (\n v -> ( n, v ))
+        |> required "name" Decode.string
+        |> optional "value" Decode.string ""
+
+
+decodeTags : Decoder (Dict TagName String)
+decodeTags =
+    Decode.list decodeTagTuple
+        |> Decode.map Dict.fromList
+
+
+preferredToString : PreferredContactMethod -> String
+preferredToString prf =
+    case prf of
+        Email ->
+            "Email"
+
+        Phone ->
+            "Phone"
+
+        Post ->
+            "Post"
+
+
+fromIdToString : Ids -> String
+fromIdToString ids =
+    case ids of
+        PrepId ->
+            "preparation"
+
+        CloseId ->
+            "closed"
+
+        CancelId ->
+            "canceled"
+
+        TimeOutId ->
+            "timedOut"
+
+        RejectId ->
+            "rejected"
+
+        ErrId ->
+            "error"
+
+        EmailId ->
+            "email"
+
+        PhoneId ->
+            "phone"
+
+        CompanyId ->
+            "companyName"
+
+        PostId ->
+            "address"
+
+        ZipId ->
+            "zip"
+
+        CityId ->
+            "city"
+
+        NewTagNameId ->
+            "newTagName"
+
+        NewTagValId ->
+            "newTagValue"
+
+        CountryId ->
+            "countryName"
+
+
+
+---- VIEW ----
 
 
 view : Model -> Html Msg
-view _ =
-    Html.div [ Attrs.class "flex flex-col w-[1024px] items-center mx-auto mt-16 mb-48" ]
-        [ header "Let's start your task"
-        , subheader "Here are your data:"
-        , Html.pre [ Attrs.class "my-8 py-4 px-12 text-sm bg-slate-100 font-mono shadow rounded" ] [ Html.text Data.userGroup ]
-        , header "Now turn them into form."
-        , subheader "See README for details of the task. Good luck 🍀 "
+view model =
+    Html.div [ Attr.class "flex flex-col sm:w-full md:w-[626px] items-center mx-auto" ]
+        [ case model.userGroup of
+            Initial ->
+                text ""
```

Comment:
It seems that Initial is actually never used. Is there any reason to have it or would you rather remove it completely?

Reply:
Given that "get" request is triggering on init, it makes a lot of sense to remove it completely 👍